lib/TrendApplication.php

```diff
@@ -95,4 +142,42 @@ private function compare($key, $referenceValue, $comparingValue, $exitCode): int
         }
         return $exitCode;
     }
+
+    /**
+     * @param ResultPrinter::KEY_* $key
+     * @param int $referenceValue
+     * @param int $comparingValue
+     * @param self::EXIT_* $exitCode
+     *
+     * @return self::EXIT_*
+     */
+    private function compareForJson(OutputResultJson $result, $key, $referenceValue, $comparingValue, $exitCode): int
```

Comment:
I don't like duplicating most logic from `compare()`.

please refactor and separate logic from output instead.